syncapi/storage/postgres/syncserver.go

```diff
@@ -101,9 +101,14 @@ func NewDatabase(base *base.BaseDendrite, dbProperties *config.DatabaseOptions)
 	m := sqlutil.NewMigrations()
 	deltas.LoadFixSequences(m)
 	deltas.LoadRemoveSendToDeviceSentColumn(m)
+	deltas.LoadAddHistoryVisibilityColumn(m)
```

Comment:
Oh, good catch!

Reply:
Hm.. I think the ordering is correct. We first create the table (separate `db.Exec` statement at the top of `NewDatabase`), execute the migrations and prepare the statements afterwards. If we move `New*EventsTable` to be executed before the migrations, it will fail because `history_visibilty` is missing (at least when coming from earlier versions).